src/main/java/se/sundsvall/oepintegrator/service/InstanceService.java

```diff
@@ -54,22 +54,14 @@ public void updateInstance(final String municipalityId, final String id, final I
 		}
 
 		instanceRepository.save(InstanceMapper.updateInstance(entity, instance, encryptedPassword));
-		clientFactory.removeClient(municipalityId, id);
+		clientFactory.removeClient(municipalityId, instance.getInstanceType());
 		clientFactory.createClient(entity);
 	}
 
 	public void deleteInstance(final String municipalityId, final String instanceId) {
-		instanceExist(municipalityId, instanceId);
+		final var entity = instanceRepository.findById(instanceId).orElseThrow();
```

Comment:
Add `() -> Problem.valueOf(NOT_FOUND, "Instance with id %s is not found".formatted(instanceId)` to orElseThrow?